src/pe/export.rs

```diff
@@ -195,47 +222,69 @@ impl<'a, 'b> scroll::ctx::TryFromCtx<'a, ExportCtx<'b>> for Export<'a> {
     #[inline]
     fn try_from_ctx(bytes: &'a [u8], ExportCtx { ptr, idx, sections, addresses, ordinals }: ExportCtx<'b>) -> Result<(Self, Self::Size), Self::Error> {
         use self::ExportAddressTableEntry::*;
-        let i = idx;
-        let name_offset = utils::find_offset_or(ptr as usize, sections, &format!("Cannot map export name pointer rva ({:#x}) into offset", ptr))?;
-        let name = bytes.pread::<&str>(name_offset)?;
-        let ordinal = ordinals[i];
-        let address_index = ordinal as usize;
-        debug!("name: {} name_offset: {:#x} ordinal: {} address_index: {}", name, name_offset, ordinal, address_index);
-        if address_index >= addresses.len() {
-            debug!("Export.pread bad address index ({}) for {}: idx: {} ordinal: {} addresses.len(): {}", name, address_index, i, ordinal, addresses.len());
-            Ok((Export::default(), 0))
-        } else {
-            match addresses[address_index] {
-                ExportRVA(rva) => {
-                    let rva = rva as usize;
-                    let offset = utils::find_offset_or(rva, sections, &format!("Cannot map export rva ({:#x}) into offset for {}", rva, name))?;
-                    debug!("{}: {:#x}", name, offset);
-                    Ok((Export { name: name, offset: offset, rva: rva, reexport: None, size: 0 }, 0))
-                },
-                ForwarderRVA(rva) => {
-                    let rva = rva as usize;
-                    let offset = utils::find_offset_or(rva, sections, &format!("Cannot map forwarder rva ({:#x}) into offset for {}", rva, name))?;
-                    debug!("{}: stroffset {:#x}", name, offset);
-                    let reexport = Reexport::parse(bytes, offset)?;
-                    debug!("{}: reexport {:?}", name, reexport);
-                    Ok((Export { name: name, offset: rva, rva: rva, reexport: Some(reexport), size: 0 }, 0))
-                },
+        
+        let name = ptr.map_or(None, |ptr| {
+            let name_offset = utils::find_offset(ptr as usize, sections);
+            name_offset.and_then(|offset| bytes.pread::<&str>(offset).ok())
+        });
+
+        let ordinal = ordinals.get(idx);
```

Comment:
This check should be in the loop below which constructs the export vector; not here, this way you don't have to make every field in export None; I believe your switching name to None is valid, but the other fields all cannot be, for the reasons I gave above.